tutorial/nixify/1_nix_flake.md

```diff
@@ -0,0 +1,361 @@
+      };
+    };
+}
+```
+Let's break it down!
+### haskellPackages
+
+Consult [the official manual](https://nixos.org/manual/nixpkgs/unstable/#haskell) to learn more about the Haskell infrastructure in nixpkgs, but for the purpose of our blog post it is suffice to know that:
+
+- `pkgs.haskellPackages` is an attribute set that contains all Haskell packages maintained within `nixpkgs`.
+- Since our local package (`todo-app`) is not already included in `pkgs.haskellPackages`, we need to manually add it.
+- Technically, you can use `packages.${system}.default = pkgs.${system}.haskellPackages.callCabal2nix "todo-app" ./. { };` to include the package. However, adding it to `haskellPackages` consolidates every Haskell package in one place. 
+
+In summary, adding the local package to `pkgs.haskellPackages` centralizes the package management process and simplifies the usage of the package within other flakes.
+
+### Overlay
+
+[Overlays](https://nixos.wiki/wiki/Overlays) are used to override an existing package set, such as `pkgs.haskellPackages`, and produce a new package set containing the changes. These changes could be either about overriding a single package in the package set (the second argument `super` references the original package set), or it could be about adding new packages to it. 
+
+### callCabal2nix
+
+The `callCabal2nix` function generates a Haskell package derivation based on its source. This function internally utilizes ["cabal2nix"](https://github.com/NixOS/cabal2nix), which is a Haskell utility that generates Nix build instructions from a cabal file.
+
+
+### Time to run!
+[![asciicast](https://asciinema.org/a/1dRgdxS6PfOPBoKpzrDxGa17p.svg)](https://asciinema.org/a/1dRgdxS6PfOPBoKpzrDxGa17p)
+
+## Nixify DevShell
+
+Our existing flake enables us to *build* the `todo-app`. However, what if want to develop it, by adding a feature or fixing a bug? For Haskell development, we normally use [cabal](https://cabal.readthedocs.io/) and tools like [ghcid](https://github.com/ndmitchell/ghcid). These tools require a GHC environment that includes the packages specified in the `build-depends` of our cabal file. This is where `devShell` becomes useful as it provides an isolated environment with all packages required by the project, catering to our development needs.
+
+Tl;dr Here is the `flake.nix` for this section:
+
+```nix
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  };
+  outputs = { self, nixpkgs }:
+    let
+      system = "aarch64-darwin";
+      pkgs = nixpkgs.legacyPackages.${system};
+      overlay = final: prev: {
+        todo-app = final.callCabal2nix "todo-app" ./. { };
+      };
+      myHaskellPackages = pkgs.haskellPackages.extend overlay;
+    in
+    {
+      devShells.${system}.default = myHaskellPackages.shellFor {
+        packages = p : [
+          p.todo-app
+        ];
+        buildInputs = with myHaskellPackages; [
+          ghcid
+          cabal-install
+          todo-app
```

Comment:
If you put `todo-app` in shell, running `nix develop` will recompile the project every time the .cabal or .hs file is changed.

Reply:
I'd remove it from the post, but you don't have to re-create the asciinema.